owais635/planner_flutter
# add activity timings to courses

Issue Reference: #17

### Goals
- This PR adds timing information to Courses so that we can then implement Timetable view in #6.

### Details
- This PR adds "Activities" to Courses.
- Use Tabs in Course Details Screen.
    - The tabs show Course Details and Tasks for the Course.
- Fixes a typo in `lib/models/subtask.dart`.
- Mark some issues with `TODO`.

### Manual Testing
- Testing Backwards Compatibility:
    - Before checking out this branch, create a few Courses.
    - Then checkout this branch.
    - Go to Course Details Screen.
        - You should be able to view the old course's details and their tasks.
        - You should be able to modify the Courses (add activities, edit colour, etc).
- Testing Activities
    - Add, update, delete Activities for Courses.

<img src="https://user-images.githubusercontent.com/25351744/126025341-578a1f86-5f7c-457c-8472-e4568b8d0dbe.png" height="500">

<img src="https://user-images.githubusercontent.com/25351744/126025364-a9881f35-c10d-47a5-9042-b4bc3d859f89.png" height="500">

<img src="https://user-images.githubusercontent.com/25351744/126025372-4323feed-de80-418a-ae07-9af75eada4c5.png" height="500">
